dali/imgcodec/decoders/nvjpeg/nvjpeg_memory.cc

```diff
@@ -0,0 +1,156 @@
+      case NVJPEG2K_STATUS_INTERNAL_ERROR:
+        return "Unknown error occurred in the library.";
+      case NVJPEG2K_STATUS_IMPLEMENTATION_NOT_SUPPORTED:
+        return "API is not supported by the backend.";
+      default:
+        return "< unknown error >";
+    }
+  }
+
+  static std::string Message(nvjpeg2kStatus_t result, const char *details) {
+    if (details && *details) {
+      return make_string("nvJPEG2000 error (", result, "): ", ErrorString(result),
+                         "\nDetails:\n", details);
+    } else {
+      return make_string("nvJPEG2000 error (", result, "): ", ErrorString(result));
+    }
+  }
+
+  nvjpeg2kStatus_t result() const { return result_; }
+
+ private:
+  nvjpeg2kStatus_t result_;
+};
+
+struct NvJpeg2kHandle : public UniqueHandle<nvjpeg2kHandle_t, NvJpeg2kHandle> {
+  DALI_INHERIT_UNIQUE_HANDLE(nvjpeg2kHandle_t, NvJpeg2kHandle);
+
+  NvJpeg2kHandle() = default;
+
+  NvJpeg2kHandle(nvjpeg2kDeviceAllocator_t *dev_alloc, nvjpeg2kPinnedAllocator_t *pin_alloc) {
+    if (nvjpeg2kCreate(NVJPEG2K_BACKEND_DEFAULT, dev_alloc, pin_alloc, &handle_) !=
+        NVJPEG2K_STATUS_SUCCESS) {
+      handle_ = null_handle();
+    }
+  }
+
+  static constexpr nvjpeg2kHandle_t null_handle() { return nullptr; }
+
+  static void DestroyHandle(nvjpeg2kHandle_t handle) {
+    nvjpeg2kDestroy(handle);
+  }
+};
+
+struct NvJpeg2kStream : public UniqueHandle<nvjpeg2kStream_t, NvJpeg2kStream> {
+  DALI_INHERIT_UNIQUE_HANDLE(nvjpeg2kStream_t, NvJpeg2kStream);
+
+  static NvJpeg2kStream Create() {
+    nvjpeg2kStream_t handle{};
+    CUDA_CALL(nvjpeg2kStreamCreate(&handle));
+    return NvJpeg2kStream(handle);
+  }
+
+  static constexpr nvjpeg2kStream_t null_handle() { return nullptr; }
+
+  static void DestroyHandle(nvjpeg2kStream_t handle) {
+    nvjpeg2kStreamDestroy(handle);
+  }
+};
+
+struct NvJpeg2kDecodeState : public UniqueHandle<nvjpeg2kDecodeState_t, NvJpeg2kDecodeState> {
+  DALI_INHERIT_UNIQUE_HANDLE(nvjpeg2kDecodeState_t, NvJpeg2kDecodeState);
+
+  explicit NvJpeg2kDecodeState(nvjpeg2kHandle_t nvjpeg2k_handle) {
+    CUDA_CALL(nvjpeg2kDecodeStateCreate(nvjpeg2k_handle, &handle_));
+  }
+
+  static constexpr nvjpeg2kDecodeState_t null_handle() { return nullptr; }
+
+  static void DestroyHandle(nvjpeg2kDecodeState_t handle) {
+    nvjpeg2kDecodeStateDestroy(handle);
+  }
+};
+
+}  // namespace imgcodec
+
+template <>
+inline void cudaResultCheck<nvjpeg2kStatus_t>(nvjpeg2kStatus_t status) {
+  switch (status) {
+  case NVJPEG2K_STATUS_SUCCESS:
+    return;
+  default:
+    throw dali::imgcodec::NvJpeg2kError(status);
+  }
+}
+
+template <>
+inline void cudaResultCheck<nvjpeg2kStatus_t>(nvjpeg2kStatus_t status, const string &extra) {
+  switch (status) {
+  case NVJPEG2K_STATUS_SUCCESS:
+    return;
+  default:
+    throw dali::imgcodec::NvJpeg2kError(status, extra.c_str());
+  }
+}
+
+}  // namespace dali
+
+#endif  // DALI_IMGCODEC_DECODERS_NVJPEG_NVJPEG2K_HELPER_H_
```

Comment:
This won't pass linter, please fix the newlines and the header guard (it doesn't match the name of the file).